src/core/thread/mle.cpp

```diff
@@ -739,6 +739,7 @@ otError Mle::SetStateChild(uint16_t aRloc16)
     SetRloc16(aRloc16);
     SetRole(OT_DEVICE_ROLE_CHILD);
     SetAttachState(kAttachStateIdle);
+    mAttachTimer.Stop();
```

Comment:
Together with #3537 , is it possible to stop in `SetAttachState()` when the state is `kAttachStateIdle`?